app/controllers/posts_controller.rb

```diff
@@ -6,6 +6,27 @@ def index
 
   def show
     @user = User.find(params[:user_id])
-    @post = Post.find(params[:id])
+    @post = @user.posts.find(params[:id])
+    @like = current_user.likes.where(post_id: params[:id]).exists?
+  end
+
+  def new
+    @current_user = current_user
+  end
+
+  def create
+    new_post = current_user.posts.new(params.require(:data).permit(:title, :text))
```

Comment:
- [ ] Let's implement in a better way how we work with strong parameters.

  Fistly, we need a private method called `post_params` that will take care of encapsulating our strong params. Using a private method to encapsulate the permissible parameters is a good pattern since you'll be able to reuse the same permit list between create and update actions where necessary. But, most importantly, it provides an interface for protecting attributes from end-user assignment.
  
  So, we should have something like this:
  
  ```ruby
  def create
    @post = @current_user.posts.new(post_params)
    # the rest of the create logic
  end
  
  private 
  def post_params
    params.require(:post).permit(:title, :text)
  end
```

Reply:
@Lameck1 Done, thank you.